### PR TITLE
DSM-PEPPER-94-dont-double-click

### PR DESCRIPTION
**Pull request by GiCharkviani**
[PEPPER-94](https://broadworkbench.atlassian.net/browse/PEPPER-94)

**SOLUTION:** DIsabling checkboxes while fetching data

How it feels **like:**

https://user-images.githubusercontent.com/77500504/201079158-87982a33-08c1-4372-a2dd-dadb502e4137.mov

### CODE REVIEW

File: ddp-workspace/projects/ddp-dsm-ui/src/app/shipping/shipping.component.ts [hunk 1]
```diff
@@ -271,7 +274,9 @@ export class ShippingComponent implements OnInit {
     if (localStorage.getItem(ComponentService.MENU_SELECTED_REALM) != null &&
       localStorage.getItem(ComponentService.MENU_SELECTED_REALM) !== ''
     ) {
+      this.disableCheckboxes = true;
       this.dsmService.getKitRequests(localStorage.getItem(ComponentService.MENU_SELECTED_REALM), this.shippingPage, kitType.name)
+        .pipe(finalize(() => this.disableCheckboxes = false))
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/shipping/shipping.component.ts:279]: It would be nice to add `unsubscribe` for the subscription.

**Review by alextanlan — APPROVED** [review]

File: ddp-workspace/projects/ddp-dsm-ui/src/app/shipping/shipping.component.ts (hunk 1 above)
Comment (GiCharkviani) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/shipping/shipping.component.ts:279]: [Don't think it needs that,](https://lukaonik.medium.com/do-we-need-to-unsubscribe-http-client-in-angular-86d781522b99) as .getKitRequests() returns HttpCLient get() function ref.

Comment (alextanlan) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/shipping/shipping.component.ts:279]: Generally yes.
But there is [some nuance/alternative opinion](https://medium.com/angular-in-depth/why-you-have-to-unsubscribe-from-observable-92502d5639d0#:~:text=but%20what%20would%20be%20the%20best%20way%20to%20work%20with%20HttpClient%3F).:wink:

Comment (GiCharkviani) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/shipping/shipping.component.ts:279]: I am aware of that opinion. It's still not needed here, though. :D

Comment (alextanlan) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/shipping/shipping.component.ts:279]: no problem

**Review by SimoneMaiwald — APPROVED** [review]

**Pull request merged**